frontend/src/Components/PageNav.tsx

```diff
@@ -1,48 +1,28 @@
-import { useEffect, useRef, useState } from 'react';
+import { useEffect, useRef } from 'react';
 import { isAdministrator, useAuth } from '@/useAuth';
-import {
-    Bars3Icon,
-    BuildingOffice2Icon,
-    HomeIcon
-} from '@heroicons/react/24/solid';
+import { Bars3Icon, BuildingOffice2Icon } from '@heroicons/react/24/solid';
 import ULIComponent from '@/Components/ULIComponent.tsx';
 import { Facility } from '@/common';
-import { useLoaderData } from 'react-router-dom';
+import { useMatches, useLoaderData } from 'react-router-dom';
 import API from '@/api/api';
-import { usePathValue } from '@/Context/PathValueCtx';
+
+interface HandleType {
+    title: string;
+}
 
 export default function PageNav({
-    path,
     showOpenMenu,
     onShowNav
 }: {
-    path: string[];
     showOpenMenu: boolean;
     onShowNav?: () => void;
 }) {
     const { user } = useAuth();
     const detailsRef = useRef<HTMLDetailsElement>(null);
-    const { pathVal } = usePathValue();
-    const [customPath, setCustomPath] = useState<string[]>(path ?? []);
     const facilityNames = useLoaderData() as Facility[] | null;
-
-    useEffect(() => {
-        const handlePathChange = () => {
-            const newPath = [...path];
-            if (newPath && newPath.length > 0) {
-                setCustomPath(
-                    newPath.map((p) => {
-                        return (
-                            pathVal?.find((pv) => pv.path_id === p)?.value ?? p
-                        );
-                    })
-                );
-                return;
-            }
-            setCustomPath(path);
-        };
-        handlePathChange();
-    }, [path, pathVal]);
+    const matches = useMatches();
+    const currentRoute = matches[matches.length - 1];
+    const pageTitle = (currentRoute?.handle as HandleType)?.title;
```

Comment:
Are we able to use `TitleHandler` from `common.ts` here or do we need to use `HandleType`?